l2geth/rollup/sync_service.go

```diff
@@ -1151,13 +1156,78 @@ func (s *SyncService) verifyFee(tx *types.Transaction) error {
 	return nil
 }
 
+func (s *SyncService) VerifiedTxCount() (uint64, error) {
+	var pendingTxCount uint64
+	pendingTxs, err := s.txpool.Pending()
+	if err != nil {
+		return 0, err
+	}
+	for _, txs := range pendingTxs {
+		for _, tx := range txs {
+			if err := s.ValidateSequencerTransaction(tx, common.Address{}); err == nil {
+				pendingTxCount++
+			}
+		}
+	}
+	return pendingTxCount, nil
+}
+
+// ApplyUpdateGasPriceTxs apply tx for update gasPrice
+func (s *SyncService) ApplyUpdateGasPriceTxs() {
+	for {
+		if len(s.updateGasPriceTxPool) == 0 {
+			break
+		}
+		tx := <-s.updateGasPriceTxPool
+		if err := s.ValidateAndApplySequencerTransaction(tx, &types.BatchTxSetProof{}); err != nil {
+			log.Error("apply update gasPrice tx error", "err_msg", err.Error())
+		}
+	}
+}
+
+func (s *SyncService) IsUpdateGasPriceTx(tx *types.Transaction) bool {
+	if tx == nil {
+		return false
+	}
+	from, err := types.Sender(s.signer, tx)
+	if err != nil {
+		return false
+	}
+	gpoOwner := s.GasPriceOracleOwnerAddress()
+	if gpoOwner != nil {
+		if from != *gpoOwner {
+			return false
+		}
+	}
+	return true
+}
+
+// AddUpdateGasPriceTx add updateGasPriceTx to queue
+func (s *SyncService) AddUpdateGasPriceTx(tx *types.Transaction) error {
+	if len(s.updateGasPriceTxPool) == gasPriceTxPoolSize {
+		return fmt.Errorf("there are too many transactions waiting to be processed")
+	}
+	s.updateGasPriceTxPool <- tx
+	return nil
+}
+
 func (s *SyncService) ValidateSequencerTransaction(tx *types.Transaction, sequencer common.Address) error {
 	if s.verifier {
 		return errors.New("Verifier does not accept transactions out of band")
 	}
 	if tx == nil {
 		return errors.New("nil transaction passed to ValidateAndApplySequencerTransaction")
 	}
+	l1GasPrice, err := s.RollupGpo.SuggestL1GasPrice(context.Background())
+	if err != nil {
+		return err
+	}
+	lastL1GasPrice, ok := s.verifiedMap[tx.Hash()]
+	if ok {
+		if l1GasPrice.Cmp(lastL1GasPrice) < 1 {
```

Comment:
```
// Cmp compares x and y and returns:
//
//	-1 if x <  y
//	 0 if x == y
//	+1 if x >  y
func (x *Int) Cmp(y *Int) (r int) {
```
Why compare with `1`?

Reply:
when current l1GasPrice <= lastL1GasPrice,Indicates that the cost of L1Fee does not increase

Reply:
Less than `1` is confusing. How about change to `<= 0`

Reply:
ok